src/main/java/mate/controller/LoginController.java

```diff
@@ -0,0 +1,43 @@
+package mate.controller;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import mate.exception.AuthenticationException;
+import mate.lib.Injector;
+import mate.model.Driver;
+import mate.service.AuthenticationService;
+
+public class LoginController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private static final String DRIVER_ID = "driver_id";
+    private static final String NAME = "name";
+    private final AuthenticationService authenticationService
+            = (AuthenticationService) injector.getInstance(AuthenticationService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            Driver driver = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute(DRIVER_ID, driver.getId());
+            session.setAttribute(NAME, driver.getName());
+            resp.sendRedirect("/");
+        } catch (AuthenticationException e) {
+            req.setAttribute("error", e.getMessage());

```

Comment:
This will be more informative attribute
```suggestion
           req.setAttribute("errorMsg", e.getMessage());

```